src/test/java/com/zepben/cimbend/cim/iec61970/base/wires/PowerTransformerTest.kt

```diff
@@ -96,4 +97,30 @@ internal class PowerTransformerTest {
             .withMessage("Unable to add ${duplicatePowerTransformerEnd.typeNameAndMRID()} to ${pt.typeNameAndMRID()}. A ${e1.typeNameAndMRID()} already exists with endNumber 1.")
     }
 
+    @Test
+    internal fun `test primaryVoltage`() {
+        val pt = PowerTransformer()
+        val e1 = PowerTransformerEnd().apply {
+            powerTransformer = pt
+            baseVoltage = BaseVoltage().apply { nominalVoltage = 10 }
+            ratedU = 15
+        }
+        val e2 = PowerTransformerEnd().apply {
+            powerTransformer = pt
+            baseVoltage = BaseVoltage().apply { nominalVoltage = 20 }
+            ratedU = 25
+        }
+        pt.addEnd(e1).addEnd(e2)
+
+        assertThat(pt.primaryVoltage, equalTo(e1.baseVoltage?.nominalVoltage))
+
+        e1.baseVoltage = null
+
+        assertThat(pt.primaryVoltage, equalTo(e1.ratedU))
+
+        pt.clearEnds()
+
+        assertThat(pt.primaryVoltage, equalTo(pt.baseVoltageValue))
```

Comment:
I would suggest re-ordering this test so that it checks as you add things. i.e.

```
val pt = PowerTransformer()
val e1 = PowerTransformerEnd().apply{
    powerTransformer = pt
}
val e2 = PowerTransformerEnd().apply {
    powerTransformer = pt
    baseVoltage = BaseVoltage().apply { nominalVoltage = 20 }
    ratedU = 25
}

assertThat(pt.primaryVoltage, equalTo(0))

pt.baseVoltage = BaseVoltage().apply { nominalVoltage = 5 }
assertThat(pt.primaryVoltage, equalTo(5))

pt.addEnd(e1).addEnd(e2)
assertThat(pt.primaryVoltage, equalTo(0))

e1.ratedU = 15
assertThat(pt.primaryVoltage, equalTo(15))

e1.baseVoltage = BaseVoltage().apply { nominalVoltage = 10 }
assertThat(pt.primaryVoltage, equalTo(10))
```

Reply:
@roberto-marquez also note that the above check I suggested includes an end with no voltage